Analysis/interface/TauSelection.h

```diff
@@ -3,7 +3,7 @@
 #include "TauMLTools/Analysis/interface/GenLepton.h"
 #include "TauMLTools/Analysis/interface/AnalysisTypes.h"
 #include "TauMLTools/Core/interface/AnalysisMath.h"
-#include "TauMLTools/Core/interface/exception.h"
+#include "TauMLTools/Core/src/exception.hpp"
```

Comment:
please keep the name as `TauMLTools/Core/interface/exception.h`, because including files from src looks unusual. Also like this you don't need to modify all the files where it is used

Reply:
If I keep the file in interface, some other files (e.g. Production/src/GenTruthTools.cc) are missing the definition of analysis::exception. I believe that scram compiles into shared libraries only what is under src/
As a solution, we can leve exception under interface/exception.h and explicitly include exception.h in all these files

Reply:
if the whole class is declared in .h, there shouldn't be problems with missing definitions. Maybe you need to run `scram b clean`?

Reply:
indeed, it compiles now